### PR TITLE
Migration Docs: Restructure plugin migration docs

### PR DESCRIPTION
**Pull request by leventebalogh**
ℹ️ **Actual work was done by @sympatheticmoose.**

### What changed?
Restructured the documentation around plugin migration to make relevant sections easier to find and also to make a place for the angular migration guides. (The structure is mostly based on the https://grafana.com/docs/grafana/latest/upgrade-guide).

- [x] Extract the angular migration guide to it's own page
- [x] Extract the different plugin migration guides to their own pages

```bash
# Before
# --------------------------------------
> Developers
   - ...
   > Build a plugin
      - ...
      - Plugin migration guide # (holds both Angular and general plugin migration guide information)
   

# After
# --------------------------------------
> Developers
   - ...
   > Build a plugin
      - ...
       > Migration guides
          - Migrating plugins from AngularJS to React
          - Migrating plugins from Grafana 9.3.x to 9.4.x
          - Migrating plugins from Grafana 9.1.x to 9.2.x
          - Migrating plugins from Grafana 8.3.x to 8.4.x
          - Migrating plugins from Grafana 8.x to 9.x
          - Migrating plugins from Grafana 7.x to 8.x
          - Migrating plugins from Grafana 6.x to 7.x
   
```

### CODE REVIEW

**Comment by sympatheticmoose** [comment]
as things stand, the redirects don't appear to be working as far as I can tell locally

**Comment by sympatheticmoose** [comment]
@leventebalogh over to you!

File: docs/sources/developers/plugins/migration-guide/angular-react/_index.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+---
+
+# Migrate a plugin from AngularJS to React
+
+If you’re looking to migrate a plugin to the new plugin platform, then we recommend that you release it under a new major version. Consider keeping a release branch for the previous version to be able to roll out patch releases for versions prior to Grafana 7.
+
+While there's no 1-to-1 migration path from an Angular plugin to the new React platform, from early adopters, we’ve learned that one of the easiest ways to migrate is to:
+
+1. Create a new branch called `migrate-to-react`.
+1. Start from scratch with one of the templates provided by Grafana Toolkit.
```
Comment (sympatheticmoose) [inline docs/sources/developers/plugins/migration-guide/angular-react/_index.md:21]: @leventebalogh just checking - are we keeping restructuring and updates in separate PRs? I'd expect we need to update the content away from Toolkit etc

Comment (leventebalogh) [reply docs/sources/developers/plugins/migration-guide/angular-react/_index.md:21]: Yeah, I would ideally keep this one only for the restructuring (and for small changes that are necessary for the restructure, e.g. the redirect logic.)

**Review by mckn — APPROVED** [review]
LGTM!

**Pull request merged**